Lib/fontTools/varLib/builder.py

```diff
@@ -26,39 +26,40 @@ def buildVarRegionList(supports, axisTags):
 	return self
 
 
-def _reorderItem(lst, narrows, zeroes):
-	out = []
-	count = len(lst)
-	for i in range(count):
-		if i not in narrows:
-			out.append(lst[i])
-	for i in range(count):
-		if i in narrows  and i not in zeroes:
-			out.append(lst[i])
-	return out
+def _reorderItem(lst, mapping):
+	return [lst[i] for i in mapping]
 
 def VarData_calculateNumShorts(self, optimize=False):
 	count = self.VarRegionCount
 	items = self.Item
-	narrows = set(range(count))
-	zeroes = set(range(count))
+	bit_lengths = [0] * count
 	for item in items:
-		wides = [i for i in narrows if not (-128 <= item[i] <= 127)]
-		narrows.difference_update(wides)
-		nonzeroes = [i for i in zeroes if item[i]]
-		zeroes.difference_update(nonzeroes)
-		if not narrows and not zeroes:
-			break
+		bl = [(i + (i < -1)).bit_length() for i in item]
```

Comment:
is `i + (i < -1)` maybe too clever a trick? you're deliberately using the fact that True/False coerce to integers 1 and 0 in python, but it makes the code slightly harder to understand (to me at least)

Reply:
gotcha